### DISCUSSION TITLE
Media Card

### DISCUSSION THREAD

**Discussion by bushbrother** [cdnninja/yoto_ha#47] [General]
Hi, I have been following in the background, I own 2 x v2 model Yoto and both are logged in and picked up fine, but not really seeing anything on the media player in HA that works? I know you are still in early dev, but wondering if the media player when added to a card should show anything? I just get it saying "On" as the status all the time and play/pause doesn't work?

**Comment by cdnninja** [comment]
In the master version play pause and stop function.  Playing media also works via service call.   Some of this isn't a released version yet so you would need to update to master branch in HACS. 

I also don't know how well this handles two players right now.  I bet the updates won't work correctly. Let me know what you are seeing.

**Reply by bushbrother** [reply]
Is master the same as the 1.9.0 release? I have that installed but nothing seems to work as you say above

**Reply by cdnninja** [reply]
Master is the core branch vs released version lag.  They auto release twice a week.

**Comment by cdnninja** [comment]
I just did a release of 1.10 to make it easy to get latest.  Could you install and let me know if controls work for you? I wouldn't doubt it if all commands and data come from a single player.

**Reply by bushbrother** [reply]
Thanks. Looks better, I get volume bar etc. But play/pause and titles don't seem to pass through, but clicking the volume bar works and i see it actioned to player, but it instantly goes back to 50% on HA yet its correctly adjusted on player. Let me know if you want logs or anything?

**Reply by cdnninja** [reply]
Let me fix my known bug on multi players and see first.  

I do find volume bar does that when not actively playing.   To see the playing item you also have to click into the entity.

**Comment by cdnninja** [comment]
I think I have fixed multi device updates in 1.10.2. Give it a try and let me know!  I do think I need to dial in that 50% volume bug.

**Reply by bushbrother** [reply]
I get an error when trying to press play/pause button, and it doesnt seem to show any status other than "on". I have redacted the key as I wasn't sure if that is sensitive or not, I think it's the player ID.

HA just pops up saying failed to call service. Log below. I tried removing the account and adding again, same error.

Last logged: 20:46:17

[139780277699776] Unexpected exception
[139780246592576] Unexpected exception
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/components/websocket_api/commands.py", line 239, in handle_call_service
    response = await hass.services.async_call(
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/core.py", line 2738, in async_call
    response_data = await coro
                    ^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/core.py", line 2779, in _execute_service
    return await target(service_call)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/helpers/service.py", line 975, in entity_service_call
    single_response = await _handle_entity_call(
                      ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/helpers/service.py", line 1047, in _handle_entity_call
    result = await task
             ^^^^^^^^^^
  File "/config/custom_components/yoto/media_player.py", line 67, in async_media_play
    await self.coordinator.async_resume_player(self.player.id)
  File "/config/custom_components/yoto/coordinator.py", line 92, in async_resume_player
    await self.hass.async_add_executor_job(
  File "/usr/local/lib/python3.12/concurrent/futures/thread.py", line 58, in run
    result = self.fn(*self.args, **self.kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/yoto_api/YotoManager.py", line 62, in resume_player
    self.mqtt_client[player_id].card_resume(deviceId=player_id)
    ~~~~~~~~~~~~~~~~^^^^^^^^^^^
KeyError: 'long_key_here_removed'

**Reply by cdnninja** [reply]
Looks like my multi device fix broke some things.  Mine too.  I'll work on it and let you know.

**Reply by cdnninja** [reply]
Actually mine still works fine.   Are you clicking in device play after you paused an existing card?   Or are you using a service.  As well to confirm in the native app does it show a card paused or just blank?  That key as long as it doesn't mention token is fine to share.  It is the device ID. 

What version are you on?

**Reply by bushbrother** [reply]
Yea it's just the device ID. I am just opening the player from within HA service and clicking the play/pause button.

I have the yoto playing a card but it doesn't show anything,  just says "on".

I am using 1.10.2

![Screenshot_20240520_213918_Home Assistant](https://github.com/cdnninja/yoto_ha/assets/11506785/b6498c00-0fce-4bca-aa47-00de532b34e0)
![Screenshot_20240520_213910_Home Assistant](https://github.com/cdnninja/yoto_ha/assets/11506785/f530384c-80ae-408c-9bc7-60f742c202b7)
![Screenshot_20240520_213903_Home Assistant](https://github.com/cdnninja/yoto_ha/assets/11506785/d9bd5ffe-5c0b-4779-bfcb-bbf018bb6a1b)

**Reply by cdnninja** [reply]
Found the bug.  New release out should fix that.

**Reply by bushbrother** [reply]
Amazing!  Working now on both players with titles! Thank You!

**Reply by kobbygl** [reply]
Where do you get the Content ID from?

**Reply by cdnninja** [reply]
> Where do you get the Content ID from?

You can browse the media now if you just want to select something.   If you want to create an automation start playing the item you want and check in the media player entity in the developer menu to see what it is.  As a note you can send just the part before the dash if you want to start at the first chapter.  

![image](https://github.com/user-attachments/assets/050763c0-4505-4077-ac09-fd6838f86b69)
